lib/Dialect/Torch/Transforms/ReifyDtypeCalculations.cpp

```diff
@@ -106,13 +106,24 @@ dtypeFunctionArgsBuilder(OpBuilder &b, Location loc,
 }
 
 namespace {
-class ReifyDtypeCalculationsPass
+struct ReifyDtypeCalculationsPass
     : public ReifyDtypeCalculationsBase<ReifyDtypeCalculationsPass> {
+  ReifyDtypeCalculationsPass() = default;
+  ReifyDtypeCalculationsPass(StringRef extraLibrary) {
+    this->extraLibrary = extraLibrary.str();
+  }
   void runOnOperation() override {
     MLIRContext *context = &getContext();
     ModuleOp module = getOperation();
     OwningOpRef<ModuleOp> library =
         parseSourceString<ModuleOp>(getAbstractInterpLibrary(), context);
+    if (!extraLibrary.empty()) {
+      OwningOpRef<ModuleOp> extraLibraryModule;
```

Comment:
duplicated in reifyshapecalulcations so it could be factor but it's only a few lines. let me know if you want it factored.

Reply:
I think the small amount of duplication is fine if you change the signature to loadExtraLibrary as I suggest below.